certbot-ci/certbot_integration_tests/nginx_tests/test_main.py

```diff
@@ -13,5 +17,41 @@ def context(request):
         integration_test_context.cleanup()
 
 
-def test_hello(context):
-    print(context.directory_url)
+@pytest.mark.parametrize('certname_pattern, params, context', [
+    ('nginx.{0}.wtf', ['run'], True),  # Passing True here makes the context
```

Comment:
This indirect parameter is pretty cool, but it certainly isn't the easiest thing to understand. One thing we could do which may make this a little more understandable is make the parameter to the context object a `dict`. The key in this case would be `default_server`. In addition to helping with readability, it allows us to easily add new parameters in the future.

The downside is lines are longer and we may never need to pass additional parameters so we're doing unnecessary work.

I don't really care either way if we implement this but I wanted to throw the idea out there.

Reply:
It suits me, making the code more self-readable. Implemented.